gridt/src/app/core/settings.service.ts

```diff
@@ -1,107 +1,136 @@
+import { Identity } from './identity.model';
+import { Settings } from './settings.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+export class SettingsService {
+  /*
+   * Subscribe to this observable to ready the API.
+   */
+  public URL = "https://api.gridt.org";
+
+  constructor (
+    private http: HttpClient,
+    private auth: AuthService,
+    private secStore: SecureStorageService
+  ) {}
+
+  public error_codes = {}
+
+  /**
+   * Create ReplaySubject which yields the latest user settings.
+   */
+  private _user_settings$ = new ReplaySubject<Settings>(1);
+  /*
+   * The depth of the ReplaySubject is set to 1, which makes sure only the latest
+   * available settings are retured to the user.
+   */
+
+  // private special_pipe = pipe(
+  //   skip(1),                // Skip the local storage update
+  //   distinctUntilChanged(), // Register the server response
+  //   skip(1),                // Skip server response
+  //   filter(val => !!val ),  // Skip no changes in the server setting
+  // );
+
+  /**
+   * Transforms the user settings Subject to an Observable
+   */
+  get the_user_settings$ (): Observable<Settings> {
+    // If the user_settings change, make sure they new settings are stored
+    // in the localstorage.
+    this._user_settings$.subscribe(
+      new_settings => this.storeLocalSettings(new_settings)
+    );
+    return this._user_settings$.asObservable();
+  }
+
+  /**
+   * Determines if settings are editable or not (depends on availability server)
+   */
+  private disabler$ = new BehaviorSubject<boolean>(false);
+
+  get isDisabled$ (): Observable<boolean> {
+    return this.disabler$.asObservable();
+  }
+
+  /**
+   * Observable to obtain settings from secure storage
+   */
+  private getLocalSettings = this.secStore.get$("settings") ;
+
+  /**
+   * Store settings in the secure storage.
+   * @param settings Settings to be stored.
+   */
+  private storeLocalSettings(settings: Settings): void {
+    console.log(`Storing ${JSON.stringify(settings)} into the local storage.`);
+    this.secStore.set$("settings", settings).subscribe();
+  }
+
+  /**
+   * Function to simulate some non-empty local storage, Only for testing purposes.
+   */
+  public populateStorage(): void {
+    console.log("Populating secure storage.");
+    let ID = {
+      id: 1,
+      username: "username",
+      bio: "Bio",
+      email: "email"};
+    this.secStore.set$("settings", {identity: ID}).subscribe();
+  }
+
+  /**
+   * Observable to obtain settings from server
+   */
+  private getServerSettings = this.auth.readyAuthentication$.pipe(
+      flatMap((options) => this.http.get<Identity>(
+        `${this.URL}/identity`,
+        options
+      )),
+      catchError( this.handleBadAuth(this.disabler$) ),
+      map( id => {
+        return {identity: id}
+      })
+    )

```

Comment:
In my opinion this would be more suitable in api.service.ts and then imported here, but maybe api.service.ts should be changed instead.

Reply:
Sure, how to treat the disabler observable? Also move that to the api.service?

Reply:
I think that that should go to settings.page.ts, because it is specifically about which UI elements and that should be handled by the page itself.

Reply:
I plan to check against the disabler also when trying to store settings locally. Especially when other pages will also be able to change settings, the disabler function should stay inside the settings.service.